resources/imports/studentFutureAbsence.yml

```
@@ -0,0 +1,92 @@
+details:
+  type: studentFutureAbsence
+  category: Attendance
+  name: Future Absences
+  table: gibbonAttendanceLogPerson
+  modes:
+    insert: true
+    update: false
+
+access:
+  module: Attendance
+  action: Set Future Absence
+
+primaryKey: gibbonAttendanceLogPersonID
+
+table:
+  gibbonPersonID:
+    name: "Student"
+    desc: "Username"
+    args:
+      filter: nospaces
+      required: true
+    relationship:
+      table: gibbonPerson
+      key: gibbonPersonID
+      field: username
+
+  date:
+    name: "Date"
+    desc: "YYYY-MM-DD"
+    args:
+      filter: date
+      required: true
+
+  gibbonAttendanceCodeID:
+    name: "Attendance Code (short, e.g. PS, A, etc.)"
+    desc: "Looks up the code ID"
```

Wherever possible, try to re-use the same strings from Gibbon, as this prevents creating new translation strings (of which there are already several thousand 😅 ) Especially for the description fields, which some of them are automatically generated by the table and the args.

```suggestion
    name: "Attendance Status"
    desc: "Short Name"
```